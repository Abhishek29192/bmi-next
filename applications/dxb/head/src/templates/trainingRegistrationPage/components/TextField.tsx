import DefaultTextField from "@bmi-digital/components/text-field";
import { microCopy } from "@bmi/microcopies";
import React, { useMemo, useState } from "react";
import { useSiteContext } from "../../../components/Site";
import { StyledCancelIcon, StyledIconButton } from "./TextFieldStyles";
import type { TextFieldProps } from "@bmi-digital/components/text-field";
import type { WithFormControlProps } from "@bmi-digital/components/form";

export type Props = TextFieldProps & WithFormControlProps<string>;

const TextField = ({ defaultValue = "", isTextArea, ...props }: Props) => {
  const { getMicroCopy } = useSiteContext();
  const [isActive, setIsActive] = useState(false);
  const [value, setValue] = useState<string>(defaultValue);

  const rightAdornment = useMemo(() => {
    if (!isActive && value && !isTextArea) {
      return (
        <StyledIconButton
          variant="text"
          data-testid="reset-input-value-icon-button"
          onClick={() => setValue(defaultValue)}
          accessibilityLabel={getMicroCopy(microCopy.CLEAR_LABEL)}
        >
          <StyledCancelIcon />
        </StyledIconButton>
      );
    }
  }, [value, isActive, defaultValue, isTextArea, getMicroCopy]);

  return (
    <DefaultTextField
      {...props}
      isTextArea={isTextArea}
      onClick={() => setIsActive(true)}
      onBlur={() => setIsActive(false)}
      onChange={setValue}
      rightAdornment={rightAdornment}
      defaultValue={defaultValue}
      value={value}
    />
  );
};

export default TextField;
