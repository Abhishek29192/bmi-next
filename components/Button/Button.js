"use strict";
exports.__esModule = true;
var react_1 = require("react");
var Button = function () { return <div>hello</div>; };
