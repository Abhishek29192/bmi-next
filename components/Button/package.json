{
  "name": "@bmi/button",
  "version": "1.0.0",
  "dependencies": {
    "react": "^16.13.1"
  }
}
