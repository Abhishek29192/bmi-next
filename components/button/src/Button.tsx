import React from "react";
import MaterialButton, {
  ButtonProps as MuiButtonProps
} from "@material-ui/core/Button";
import MaterialIconButton, {
  IconButtonProps as MuiIconButtonProps
} from "@material-ui/core/IconButton";
import styles from "./Button.module.scss";
import classnames from "classnames";
import { withClickable } from "@bmi/clickable";

type Variant = "text" | "outlined" | "contained";

export type IconButtonProps = Omit<MuiIconButtonProps, "action"> & {
  isIconButton: true;
  accessibilityLabel: string;
  hasDarkBackground?: boolean;
  variant?: Variant;
  // TODO: Use numbers for all the options.
  size?: "extra-small" | "small" | 42 | "medium" | "large" | "extra-large";
  component?: undefined;
};

export type ButtonProps = Omit<MuiButtonProps, "action"> & {
  isIconButton?: false;
  accessibilityLabel?: string;
  hasDarkBackground?: boolean;
  variant?: Variant;
  component?: React.ElementType<any>;
};

const Button = ({
  children,
  className,
  color = "primary",
  variant = "contained",
  hasDarkBackground,
  isIconButton,
  accessibilityLabel,
  size,
  disabled,
  component,
  ...rest
}: ButtonProps | IconButtonProps) => {
  return isIconButton ? (
    <MaterialIconButton
      className={classnames(
        styles["IconButton"],
        styles[`IconButton--${size || "medium"}`],
        {
          [styles[`IconButton--${variant}`]]: variant,
          [styles["IconButton--disabled"]]: disabled,
          [styles["IconButton--dark-background"]]: hasDarkBackground
        },
        className
      )}
      aria-label={accessibilityLabel}
      {...rest}
    >
      {children}
    </MaterialIconButton>
  ) : (
    <MaterialButton
      className={classnames(styles["Button"], className, {
        [styles["Button--dark-background"]]: hasDarkBackground
      })}
      variant={variant}
      color={color}
      size={size}
      component={component}
      disabled={disabled}
      {...rest}
    >
      {children}
    </MaterialButton>
  );
};

export default withClickable<ButtonProps | IconButtonProps>(Button);
