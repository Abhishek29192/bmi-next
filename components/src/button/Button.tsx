import React from "react";
import {
  Button as MaterialButton,
  ButtonProps as MuiButtonProps
} from "@material-ui/core";
import {
  IconButton,
  IconButtonProps as MuiIconButtonProps
} from "@material-ui/core";
import classnames from "classnames";
import { withClickable } from "../clickable/Clickable";
import { useButtonStyles, useIconButtonStyles } from "./styles";

type Variant = "text" | "outlined" | "contained" | "opaqueOutlined";

export type IconButtonProps = Omit<MuiIconButtonProps, "action"> & {
  isIconButton: true;
  accessibilityLabel: string;
  hasDarkBackground?: boolean;
  variant?: Variant;
  // TODO: Use numbers for all the options.
  size?: "extra-small" | "small" | 42 | "medium" | "large" | "extra-large";
  component?: undefined;
  classes?: MuiButtonProps["classes"];
};

export type ButtonProps = Omit<MuiButtonProps, "action"> & {
  isIconButton?: false;
  accessibilityLabel?: string;
  hasDarkBackground?: boolean;
  variant?: Variant;
  component?: React.ElementType<any>;
};

const Button = ({
  children,
  classes,
  color = "primary",
  variant = "contained",
  hasDarkBackground,
  isIconButton,
  accessibilityLabel,
  size,
  disabled,
  component = "button",
  ...rest
}: ButtonProps | IconButtonProps) => {
  const buttonClasses = useButtonStyles();
  const iconButtonClasses = useIconButtonStyles();
  const buttonVariant = variant === "opaqueOutlined" ? "outlined" : variant;

  return isIconButton ? (
    <IconButton
      classes={{
        root: classnames(
          iconButtonClasses.root,
          iconButtonClasses[size || "medium"],
          {
            [iconButtonClasses[`${buttonVariant}`]!]: variant !== "outlined",
            [iconButtonClasses.textDark]:
              variant === "text" && hasDarkBackground
          }
        )
      }}
      aria-label={accessibilityLabel}
      component={component}
      disabled={disabled}
      {...rest}
    >
      {children}
    </IconButton>
  ) : (
    <MaterialButton
      variant={buttonVariant}
      color={color}
      size={size}
      component={component}
      disabled={disabled}
      {...rest}
      classes={{
        ...classes,
        root: classnames(buttonClasses.root, classes?.root),
        text: classnames(buttonClasses.text, classes?.text, {
          [buttonClasses.textDarkBg]: hasDarkBackground
        }),
        contained: classnames(buttonClasses.contained, classes?.contained, {
          [buttonClasses.containedDarkBg]: hasDarkBackground
        }),
        label: classnames(buttonClasses.label, classes?.label),
        startIcon: classnames(buttonClasses.startIcon, classes?.startIcon),
        outlined: classnames(classes?.outlined, {
          [buttonClasses.opaqueOutlined]: variant === "opaqueOutlined",
          [buttonClasses.outlinedDarkBg]:
            hasDarkBackground && variant !== "opaqueOutlined",
          [buttonClasses.opaqueOutlinedDarkBg]:
            variant === "opaqueOutlined" && hasDarkBackground
        })
      }}
    >
      {children}
    </MaterialButton>
  );
};

export default withClickable<ButtonProps | IconButtonProps>(Button);
