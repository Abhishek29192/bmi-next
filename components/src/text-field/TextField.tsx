import {
  InputAdornment,
  TextField as MaterialTextField,
  TextFieldProps
} from "@material-ui/core";
import { ErrorRounded } from "@material-ui/icons";
import classnames from "classnames";
import React, { ChangeEvent } from "react";
import withFormControl from "../form/withFormControl";
import styles from "./TextField.module.scss";

type AdornmentProps =
  | {
      rightAdornment?: undefined;
      leftAdornment?: React.ReactNode;
    }
  | {
      rightAdornment?: React.ReactNode;
      leftAdornment?: undefined;
    };

export type Props = Omit<TextFieldProps, "variant" | "onChange"> & {
  variant?: "outlined" | "hybrid";
  isTextArea?: boolean;
  isRequired?: boolean;
  helperText?: string;
  errorText?: string;
  error?: boolean;
  onChange?: (value: string) => void;
} & AdornmentProps;

export const TextField = ({
  className,
  variant,
  isTextArea,
  leftAdornment,
  rightAdornment,
  id,
  error,
  helperText,
  errorText,
  onChange,
  InputProps,
  isRequired,
  ...props
}: Props) => {
  const hasAdornment = error || leftAdornment || rightAdornment;
  const inputProps = hasAdornment
    ? {
        endAdornment: (
          <InputAdornment position="end">
            {error ? (
              <ErrorRounded
                style={{ color: "red" }}
                className={styles["Error"]}
              />
            ) : null}
            {leftAdornment}
            {!error ? rightAdornment : null}
          </InputAdornment>
        ),
        ...InputProps
      }
    : InputProps;

  const handleChange = (
    event: ChangeEvent<HTMLTextAreaElement | HTMLInputElement>
  ) => onChange && onChange(event.target.value);

  return (
    <MaterialTextField
      {...props}
      required={isRequired}
      helperText={error ? errorText : helperText}
      id={id}
      error={error}
      multiline={isTextArea}
      onChange={handleChange}
      variant={variant === "hybrid" ? "filled" : "outlined"}
      className={classnames(
        styles["TextField"],
        { [styles["TextField--leftAdornment"]!]: leftAdornment },
        {
          [styles["TextField--error"]!]: error
        },
        className
      )}
      InputProps={inputProps}
    />
  );
};

export default withFormControl<Props, string>(TextField);
