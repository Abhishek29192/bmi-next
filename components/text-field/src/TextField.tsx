import React from "react";
import MaterialTextField, { TextFieldProps } from "@material-ui/core/TextField";
import styles from "./TextField.module.scss";
import classnames from "classnames";

type Props = Omit<TextFieldProps, "variant"> & {
  variant?: "outlined" | "hybrid";
};

const TextField = ({ className, variant, ...props }: Props) => (
  <MaterialTextField
    {...props}
    variant={variant === "hybrid" ? "filled" : "outlined"}
    className={classnames(styles["TextField"], className)}
  />
);

export default TextField;
