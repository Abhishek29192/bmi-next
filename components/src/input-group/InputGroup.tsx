import React from "react";
import classnames from "classnames";
import { useTheme } from "@material-ui/core";
import { useMediaQuery } from "@material-ui/core";
import { ButtonProps, IconButtonProps } from "../button/Button";
import { Props as TextFieldProps } from "../text-field/TextField";
import styles from "./InputGroup.module.scss";

// TODO: This check doesn't really work.
type Props = {
  input: React.ReactElement<TextFieldProps>;
  button: React.ReactElement<ButtonProps | IconButtonProps>;
  lockBreakpoint?: "xs" | "sm" | "md" | "lg" | false;
};

const InputGroup = ({ input, button, lockBreakpoint = "sm" }: Props) => {
  const theme = useTheme();
  const matches =
    lockBreakpoint && useMediaQuery(theme.breakpoints.up(lockBreakpoint));
  const InputComponent = React.cloneElement(input, {
    className: styles["input"]
  });
  const extraProp = button.props.isIconButton
    ? {}
    : {
        disableElevation: matches
      };

  const ButtonComponent = React.cloneElement(button, {
    className: styles["button"],
    ...extraProp
  });

  return (
    <div
      className={classnames(
        styles["InputGroup"],
        lockBreakpoint && styles[`InputGroup--locked-${lockBreakpoint}`]
      )}
    >
      {InputComponent}
      {ButtonComponent}
    </div>
  );
};

export default InputGroup;
