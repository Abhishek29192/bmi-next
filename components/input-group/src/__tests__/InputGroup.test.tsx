import React from "react";
import { render } from "@testing-library/react";
import Button from "@bmi/button";
import TextField from "@bmi/text-field";
import ArrowForwardIcon from "@material-ui/icons/ArrowForward";
import InputGroup from "../";

describe("InputGroup component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <InputGroup
        input={<TextField name="first-name" label="First name" />}
        button={<Button>Submit</Button>}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it("renders unlocked", () => {
    const { container } = render(
      <InputGroup
        lockBreakpoint={false}
        input={<TextField name="first-name" label="First name" />}
        button={<Button>Submit</Button>}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it("renders with an IconButton", () => {
    const { container } = render(
      <InputGroup
        lockBreakpoint={false}
        input={<TextField name="first-name" label="First name" />}
        button={
          <Button accessibilityLabel="Submit" isIconButton>
            <ArrowForwardIcon />
          </Button>
        }
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
