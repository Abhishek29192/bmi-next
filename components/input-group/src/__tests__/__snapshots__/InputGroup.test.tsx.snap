// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroup component renders correctly 1`] = `
<div
  class="InputGroup InputGroup--locked-sm"
>
  <div
    class="MuiFormControl-root MuiTextField-root TextField input"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink="false"
    >
      First name
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="PrivateNotchedOutline-legendLabelled-3"
        >
          <span>
            First name
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained Button button MuiButton-containedPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Submit
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`InputGroup component renders unlocked 1`] = `
<div
  class="InputGroup"
>
  <div
    class="MuiFormControl-root MuiTextField-root TextField input"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink="false"
    >
      First name
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="PrivateNotchedOutline-legendLabelled-7"
        >
          <span>
            First name
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained Button button MuiButton-containedPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Submit
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`InputGroup component renders with an IconButton 1`] = `
<div
  class="InputGroup"
>
  <div
    class="MuiFormControl-root MuiTextField-root TextField input"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink="false"
    >
      First name
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="PrivateNotchedOutline-legendLabelled-11"
        >
          <span>
            First name
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    aria-label="Submit"
    class="MuiButtonBase-root MuiIconButton-root IconButton IconButton--medium IconButton--contained button"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
